core/src/Streamly/Internal/Data/Stream/Expand.hs

```diff
@@ -581,24 +581,24 @@ concatSmapMWith combine f initial =
 -- >>> s = Stream.fromList [5,1,7,9,2]
 -- >>> generate = Stream.fromPure
 -- >>> combine = Stream.mergeBy compare
--- >>> Stream.fold Fold.toList $ Stream.concatPairsWith combine generate s
+-- >>> Stream.fold Fold.toList $ Stream.mergeMapWith combine generate s
 -- [1,2,5,7,9]
 --
 -- Note that if the stream length is not a power of 2, the binary tree composed
--- by concatPairsWith would not be balanced, which may or may not be important
+-- by mergeMapWith would not be balanced, which may or may not be important
 -- depending on what you are trying to achieve.
 --
 -- /Caution: the stream of streams must be finite/
 --
 -- /Pre-release/
 --
-{-# INLINE concatPairsWith #-}
-concatPairsWith ::
+{-# INLINE mergeMapWith #-}
+mergeMapWith ::
```

Comment:
I'm unable to relate `mergeMapWith` with the function signature.
`concatPairsWith` is relatively more intuitive.
`mergeMapWith` does not make sense to me at all.

Reply:
`concatMap` can also be called `appendMap` because it maps and then combines the streams using an `append` style operation, it basically combines the new stream with the previously aggregated result. Or we can also call it a leftAssociativeLinearFlatteningMap. The canonical example of concatMap is appending (or concatenation) of generated streams. it is a map followed by concatenation.

The flaw with the `concatPairs` name is that it does not give any idea about the `map` operation. We will have to make the name longer to add map to it. If we look at the operation, it is essentially an associativeTreeFlatteningMap. The canonical application of this operation is merging of streams (e.g. merge sort). it is a map followed by a tree merge. So I called it `mergeMap`. 

I searched if anyone else is using a similar name, and found that rxjs seems to be having the same name for a similar operation. Other more intuitive suggestions are welcome.